tools/cache.py

```diff
@@ -49,13 +50,15 @@ def acquire_cache_lock(self):
       self.prev_EM_EXCLUSIVE_CACHE_ACCESS = os.environ.get('EM_EXCLUSIVE_CACHE_ACCESS')
       os.environ['EM_EXCLUSIVE_CACHE_ACCESS'] = '1'
       logging.debug('Cache: done')
+    self.acquired_count += 1
 
   def release_cache_lock(self):
-    if not self.EM_EXCLUSIVE_CACHE_ACCESS:
+    if not self.EM_EXCLUSIVE_CACHE_ACCESS and self.acquired_count == 1:
       if self.prev_EM_EXCLUSIVE_CACHE_ACCESS: os.environ['EM_EXCLUSIVE_CACHE_ACCESS'] = self.prev_EM_EXCLUSIVE_CACHE_ACCESS
       else: del os.environ['EM_EXCLUSIVE_CACHE_ACCESS']
       self.filelock.release()
       logging.debug('Cache: PID %s released multiprocess file lock to Emscripten cache' % str(os.getpid()))
+    self.acquired_count = max(self.acquired_count - 1, 0)
```

Comment:
If we ever call `release` more than `acquire`, presumably by accident.
Nothing good is going to happen then *anyway*. This should really just be an `assert`, fixed in 1b8e5f91fc3702f7c8f98724ae6c9c7009226900